### PR TITLE
API updates

### PR DESCRIPTION
**Pull request by Sleepybear**
Reopening because I renamed my branch, which apparently broke the other PR

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <name>Cloudburst API</name>
     <groupId>org.cloudburstmc</groupId>
     <artifactId>cloudburst-api</artifactId>
-    <version>0.0.1-SNAPSHOT</version>
+    <version>0.0.3-SNAPSHOT</version>
```
Comment (lukeeey) [inline pom.xml]: just a quick note - bumping the patch is fine but I recommend not bumping the major or minor versions until the API is actually properly usable

File: src/main/java/org/cloudburstmc/api/block/Block.java [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+    Block refresh();
+
+    Level getLevel();
+
+    Chunk getChunk();
+
+    Vector3i getPosition();
+
+    default int getX() {
```
Comment (lukeeey) [inline src/main/java/org/cloudburstmc/api/block/Block.java:21]: Do we need these?

Comment (lukeeey) [inline src/main/java/org/cloudburstmc/api/block/Block.java:21]: Do we need these?

File: src/main/java/org/cloudburstmc/api/block/behavior/BlockBehavior.java [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+    public ToolType getToolType(BlockState state) {
+        return state.getType().getToolType();
+    }
+
+    public TierType getMinimalTier(BlockState state) {
+        return state.getType().getTierType();
+    }
+
+    public boolean checkTool(BlockState state, ItemStack item) {
+        val toolType = getToolType(state);
```
Comment (lukeeey) [inline src/main/java/org/cloudburstmc/api/block/behavior/BlockBehavior.java:63]: oh god...`val`

could we *at least* keep this out of the API? i blame creeperface

Comment (Sleepybear) [reply src/main/java/org/cloudburstmc/api/block/behavior/BlockBehavior.java:63]: Yes, it was creeperface that put it in there, I don't see any issue with it.

File: pom.xml (hunk 1 above)
Comment (SupremeMortal) [reply pom.xml]: I agree. Until we get this fully working with the server code, it's not worth bumping the version.

File: pom.xml [hunk 4]
```diff
@@ -43,13 +43,13 @@
     <distributionManagement>
         <repository>
             <id>releases</id>
-            <name>nukkitx-releases</name>
-            <url>https://repo.nukkitx.com/release</url>
+            <name>opencollab-releases</name>
+            <url>https://repo.opencollab.dev/maven-releases</url>
```
Comment (SupremeMortal) [inline pom.xml]: We can get rid of this in favour of the solution that we use in the [Nukkit Jenkinsfile](https://github.com/CloudburstMC/Nukkit/blob/12b389a263a0425e057f283d3698d81ff258c69b/Jenkinsfile#L23-L53)

File: src/main/java/org/cloudburstmc/api/Server.java [hunk 5]
```diff
@@ -19,4 +30,28 @@
...
+
+    int getTick();
+
+    EventManager getEventManager();
+
+    boolean getAllowFlight();
+
+    PermissionManager getPermissionManager();
+
+    String getMotd();
```
Comment (SupremeMortal) [inline src/main/java/org/cloudburstmc/api/Server.java:45]: Some of these methods could be relegated to a configuration class.

**Review by SupremeMortal — CHANGES_REQUESTED** [review]

File: src/main/java/org/cloudburstmc/api/Server.java (hunk 5 above)
Comment (Sleepybear) [reply src/main/java/org/cloudburstmc/api/Server.java:45]: That sounds like a todo later PR situation

File: pom.xml (hunk 1 above)
Comment (Sleepybear) [reply pom.xml]: Updated.

**Pull request merged**